frontend/src/app/(navfooter)/new/NewScratchForm.tsx

```diff
@@ -272,26 +293,28 @@ export default function NewScratchForm({
             </div>
 
             <div>
-                <p className={styles.label}>Compiler</p>
-                <div className={styles.compilerContainer}>
-                    <div>
-                        <span className={styles.compilerChoiceHeading}>
+                <FormLabel>Compiler</FormLabel>
+                <div className="flex cursor-default select-none items-center justify-between max-[400px]:flex-col">
+                    <div className="flex w-full flex-1 flex-col">
+                        <span className="px-2.5 py-0.5 text-[0.8rem] text-[color:var(--g800)]">
                             Select a compiler
                         </span>
                         <Select
-                            className={styles.compilerChoiceSelect}
+                            className="w-full"
                             options={compilerChoiceOptions}
                             value={compilerId}
                             onChange={setCompiler}
                         />
                     </div>
-                    <div className={styles.compilerChoiceOr}>or</div>
-                    <div>
-                        <span className={styles.compilerChoiceHeading}>
+                    <div className="flex-0 px-2 py-2 text-center text-[0.8rem] text-[color:var(--g500)] min-[400px]:px-4">
```

Comment:
Surprised we don't have a tailwind config value for this: `text-[color:var(--g500)] `, like `text-grey-500` or something. If not we should raise an issue about doing that